### PR TITLE
Fix divider.

### LINKED ISSUE #107: Divider
- Replace `---` with a horizontal rule (divider)
  - This can mainly be handled in `<Child>`
  - The `onChange` event in `<Editable>` should force a blur so the cursor thought is re-rendered. 
- When the cursor is on a divider, do not allow character insertions (only allow delete or navigate)

### PR DESCRIPTION
**Pull request by thackley**
#107

### CODE REVIEW

File: src/shortcuts.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ export const globalShortcuts = perma(() => [
         const itemsRanked = lastItemsFromContextChain(contextChain)
         const children = getChildrenWithRank(itemsRanked)
 
-        if (sigKey(cursor) === '' && children.length === 0) {
+        if ((sigKey(cursor) === '' && children.length === 0) || (sigKey(cursor) === '---' && sigRank(cursor) === 0)) {
```
Comment (raineorshine) [inline src/shortcuts.js]: I don't believe we want the `sigRank` check here. We should be able to delete the cursor no matter what rank (order in the list) it has.

File: src/shortcuts.js [hunk 2]
```diff
@@ -360,7 +360,7 @@ export const globalShortcuts = perma(() => [
   {
     id: 'cursorDown',
     name: 'Cursor Down',
-    keyboard: 'ArrowDown',
+    keyboard: { key: 'ArrowDown', meta: true },
```
Comment (raineorshine) [inline src/shortcuts.js]: These should ideally be in their own PR. Something may have happened with how we ordered the commits. Not a big deal though, so just leave for now.

File: src/shortcuts.js [hunk 3]
```diff
@@ -513,6 +513,17 @@ export const globalShortcuts = perma(() => [
...
+
+  {
+    id: 'openShortcutPopup',
+    name: 'Open Shortcut Popup',
+    description: `Open the help screen which contains the tutorials and a list of all ${ isMobile ? 'gestures' : 'keyboard shortcuts'}.`,
+    keyboard: { key: '/', meta: true},
+    exec: e => {
+      window.scrollTo(0, 0)
+      store.dispatch({ type: 'showHelper', id: 'shortcuts' })
+    }
```
Comment (raineorshine) [inline src/shortcuts.js]: Here's another commit from a different PR. Do you think you may have branched from `tutorial` instead `master`?

File: src/store.js [hunk 4]
```diff
@@ -468,8 +471,9 @@ export const appReducer = (state = initialState(), action) => {
 
     // SIDE EFFECTS: syncRemoteData, localStorage, updateUrlHistory
     existingItemChange: ({ oldValue, newValue, context, showContexts, itemsRanked, rankInContext, contextChain }) => {
-
-      if (oldValue === newValue) {
+      // second condition: exit so that when deleting a divider, it does not
+      // get merged into previous thought
+      if (oldValue === newValue || newValue === oldValue + '---') {
```
Comment (raineorshine) [inline src/store.js]: Ideally this would be handled "higher up", i.e. prevent `existingItemChange` from getting dispatched at all. The check should happen where code is merged, probably in `deleteItem`.

File: src/util.js [hunk 5]
```diff
@@ -649,6 +649,12 @@ export const getContexts = (key, data=store.getState().data) => {
     // console.error(`getContexts: Unknown key "${key}" context: ${items.join(',')}`)
     return []
   }
+
+  // return when the key is --- because divider should not have superscript
+  if (key === '---') {
+    return []
+  }
+
```
Comment (raineorshine) [inline src/util.js]: This should also happen "higher up" in order to keep functions solely focused on their purpose. i.e. `getContexts` should not have to know anything about specific thoughts. Instead, make the check in `<Superscript />`. I believe there is already a check to prevent empty thoughts from rendering a superscript. Ideally you should add to that conditional.

File: src/shortcuts.js (hunk 3 above)
Comment (thackley) [reply src/shortcuts.js]: Oh looks like I did. Good catch.

Comment (raineorshine) [reply src/shortcuts.js]: You may have to put your new `git cherry-pick` skills to use!

File: src/shortcuts.js (hunk 1 above)
Comment (thackley) [reply src/shortcuts.js]: I added that for the case when a divider is the very first thought. Is there a better place to do that check to delete it?

Comment (raineorshine) [reply src/shortcuts.js]: Ah, I see! Is there a reason we need a special case for when the divider is the very first thought? Wouldn't that be handled by the normal case? Sorry if I am misunderstanding..

FWIW, the first thought may not have a rank of 0, as ranks may start at a higher number if the 0 rank thought is deleted, not to mention ranks can be negative.
